src/sentry/static/sentry/app/views/settings/project/filtersAndSampling/filtersAndSampling.tsx

```diff
@@ -41,94 +88,103 @@ class FiltersAndSampling extends AsyncView<Props, State> {
   getDefaultState() {
     return {
       ...super.getDefaultState(),
+      errorRules: [],
       transactionRules: [],
-      project: null,
+      projectDetails: null,
     };
   }
 
   getEndpoints(): ReturnType<AsyncView['getEndpoints']> {
-    // TODO(PRISCILA): it will come soon
-    return [['', '']];
+    const {organization, project} = this.props;
+    return [['projectDetails', `/projects/${organization.slug}/${project.slug}/`]];
   }
 
   componentDidMount() {
     this.getRules();
   }
 
-  getRules() {
-    const dynamicSampling: DynamicSamplingRules = [
-      {
-        condition: {
-          operator: 'globMatch',
-          name: 'releases',
-          value: ['1.1.1', '1.1.2'],
-        },
-        sampleRate: 0.7,
-        ty: 'trace',
-      },
-      {
-        condition: {
-          operator: 'and',
-          inner: [
-            {
-              operator: 'strEqualNoCase',
-              name: 'environments',
-              value: ['dev', 'prod'],
-            },
-            {
-              operator: 'strEqualNoCase',
-              name: 'userSegments',
-              value: ['FirstSegment', 'SeCoNd'],
-            },
-          ],
-        },
-        sampleRate: 0.8,
-        ty: 'error',
-      },
-      {
-        condition: {
-          operator: 'not',
-          inner: {
-            operator: 'strEqualNoCase',
-            name: 'environments',
-            value: ['dev', 'prod'],
-          },
-        },
-        sampleRate: 0.8,
-        ty: 'error',
-      },
-      {
-        condition: {
-          operator: 'strEqualNoCase',
-          name: 'environments',
-          value: ['dev', 'prod'],
-        },
-        sampleRate: 0.8,
-        ty: 'error',
-      },
-    ] as DynamicSamplingRules;
+  componentDidUpdate(_prevProps: Props, prevState: State) {
+    if (
+      !!prevState.projectDetails &&
+      (!isEqual(prevState.errorRules, this.state.errorRules) ||
+        !isEqual(prevState.transactionRules, this.state.transactionRules))
+    ) {
+      this.submitRules();
+    }
+
+    if (!isEqual(prevState.projectDetails, this.state.projectDetails)) {
```

Comment:
Do we need isEqual here? Would a simple comparison be enough?

Reply:
I've updated the code, but for now, I will need to use the isEqual 1x.